src/dialer/index.js

```diff
@@ -180,10 +190,12 @@ class Dialer {
     const { id, multiaddrs } = getPeer(peer)
 
     if (multiaddrs) {
-      this.peerStore.addressBook.add(id, multiaddrs)
+      const filteredMultiaddrs = await all(filter(multiaddrs, async (multiaddr) => !(this.connectionManager && await this.connectionManager.gater.interceptAddrDial(id, multiaddr))))
+      this.peerStore.addressBook.add(id, filteredMultiaddrs)
```

Comment:
Should the gater be called here?
The intercept below (L198) will filter addrs that are attempting to be dialed.

Reply:
To prevent malicious multiaddrs from adding to addressBook